File: 04a.pl
#! /usr/bin/perl
use warnings;
use strict;
use feature qw{ say };

use ARGV::OrDATA;

my $count = 0;
while (<>) {
    chomp;
    my @ranges = sort { $a->[0] <=> $b->[0] || $b->[1] <=> $a->[1] }
                 map [split /-/],
                 split /,/;
    ++$count if $ranges[1][1] <= $ranges[0][1];
}
say $count;


__DATA__
2-4,6-8
2-3,4-5
5-7,7-9
2-8,3-7
6-6,4-6
2-6,4-8

File: 04a.py
#! /usr/bin/python3
import sys
from operator import itemgetter

count = 0
with open(sys.argv[1], 'r') as fh:
    for line in fh:
        ranges = list(map(lambda r: list(map(int, r.split('-'))),
                          line.rstrip('\n').split(',')))
        ranges.sort(key=itemgetter(1), reverse=True)
        ranges.sort(key=itemgetter(0))
        if ranges[1][1] <= ranges[0][1]:
            count += 1
print(count)
